Feature/use snack bar

ponche opened the pull request:
Création du hook useSnackBar qui renvoie une fonction afin d'afficher la communSnackBar pendant un brief instant. 

la fonction displaySnackbar prend en paramètre, les props du composant SnackBar de la librarie React-Native-Paper 
https://callstack.github.io/react-native-paper/docs/components/Snackbar

a l'exception de la propriété onDismiss qui est devenu facultative, et la propriété visible que n'existe plus sur le type CustomSnackBarProps.

ponche committed:
👔 Création du hook useSnackBar

ce hook permet d'afficher ce snackbar pendant une bref période

ponche committed:
💄 Mise en place du SnackBar commun

une SnackBar pourvant etre affiché partout dans l'application, en utilisant le hook useSnackaBar

ponche committed:
♻️ Refactorisation de la SnackBar de ParamScreen

refactorisation de la SnackBar de ParamaScreen afin qui utilise le SnackBar commun

ponche committed:
♻️ Add action par défaut "fermer" pour useSnackbar

ajout l'action par défaut fermer avec le label pour le snackbar si non renseigné.

ponche committed:
📝 Ajout de la JsDoc du hook useSnackBar

ponche committed:
⚰️ Retire import inutile du Snackbar

jeanCodeUnMax commented on components/MainSnackBar.tsx:
```diff
@@ -0,0 +1,25 @@
+export default function MainSnackBar() {
+  const snackBar = useRecoilValue(snackBarState);
+
+  console.log(snackBar);
```
a ton besoin de la console  ?

jeanCodeUnMax commented on store/storeAtom.ts:
```diff
@@ -3,29 +3,38 @@ import { atom, selector } from "recoil";
+
+// CommunSnackBar
+export const snackBarState = atom<SnackbarProps>({
+  key: "snackBarStarte",
```
une tarte au fraise  ?

jeanCodeUnMax reviewed (commented):
akabar dabadra

jeanCodeUnMax reviewed (commented):
paramanormal

ponche committed:
✏️ Faute de frappe sur la key de atom snackBarState

ponche committed:
🔇 Retire le console.log dans MainSnackBar

ponche committed:
:twisted_rightwards_arrows: Merge remote-tracking branch 'origin/develop' into feature/useSnackBar

ponche committed:
🔧 Configuration de Jest et React Native Testing

configuration de Jest pour inclure les matchers de la library React Native Testing avec l'ajout des types

ponche committed:
➕ Ajout de ts-node

ponche committed:
🚧 Test en cours, passage des Props Snackbar Ok
